### PR TITLE
chore: Update translate to patches

### PR DESCRIPTION
**Pull request by ThomasK33**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves ENG-4775

**Please provide a short message that should be published in the vcluster release notes**
Renamed translate field to patches

**What else do we need to know?**

### CODE REVIEW

File: chart/values.schema.json [hunk 1]
```diff
@@ -3161,12 +3161,12 @@
...
           "description": "Enabled defines if this option should be enabled."
         },
-        "translate": {
+        "patches": {
           "items": {
             "$ref": "#/$defs/TranslatePatch"
           },
           "type": "array",
-          "description": "Translate the patch according to the given patches."
+          "description": "Patches the resources according to the given patches."
```
Comment (rmweir) [inline chart/values.schema.json:3169]: This is ambiguous, maybe keep "translate" here?

File: config/config.go [hunk 2]
```diff
@@ -494,8 +494,8 @@ type SyncToHostCustomResource struct {
 	// Enabled defines if this option should be enabled.
 	Enabled bool `json:"enabled,omitempty"`
 
-	// Translate the patch according to the given patches.
-	Translate []TranslatePatch `json:"translate,omitempty"`
+	// Patches the resources according to the given patches.
+	Patches []TranslatePatch `json:"patches,omitempty"`
 }
 
 type TranslatePatch struct {
```
Comment (rmweir) [inline config/config.go:501]: Do we want to rename this to just "Patch"?

File: chart/values.schema.json [hunk 3]
```diff
@@ -3047,12 +3047,12 @@
...
           "description": "RewriteHosts is a special option needed to rewrite statefulset containers to allow the correct FQDN. virtual cluster will add\na small container to each stateful set pod that will initially rewrite the /etc/hosts file to match the FQDN expected by\nthe virtual cluster."
         },
-        "translate": {
+        "patches": {
           "items": {
             "$ref": "#/$defs/TranslatePatch"
           },
           "type": "array",
-          "description": "Translate the patch according to the given patches."
+          "description": "Patches the resource according to the given patches."
```
Comment (rmweir) [inline chart/values.schema.json:3055]: This is ambiguous, maybe keep "translate" here?

**Review by rmweir — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 2 above)
Comment (deniseschannon) [reply config/config.go:501]: No, it's an array and multiple patches can be used.

File: chart/values.schema.json (hunk 3 above)
Comment (deniseschannon) [reply chart/values.schema.json:3055]: Patch is a common word used to do exactly what we're doing which is update/change a resource. 

https://kubernetes.io/docs/reference/kubectl/generated/kubectl_patch/

File: config/config.go (hunk 2 above)
Comment (rmweir) [reply config/config.go:501]: I'm referring to the struct not the field name. There is also the struct ` EnableSwitchWithTranslate` which should probably be changed to `EnabledSwitchWithPatch`.

**Review by rmweir — COMMENTED** [review]
1. I think everything prefixed with `Translate` can have that removed or swapped for `Patch`.
2. I think `EnableSwitchWithTranslate` especially need to be changed to `EnabledSwitchWithPatch`.
3. Patch can both be a noun and a verb but I don't think it should be both in a single sentence. I would prefer a description such as "Defines patches to be applied to resources." or "Defines patches for resources", as patches are _usually_ used for one thing- patching.

File: chart/values.schema.json (hunk 1 above)
Comment (ThomasK33) [reply chart/values.schema.json:3169]: How would I keep it here, given that this file gets generated off the Go struct?

File: config/config.go (hunk 2 above)
Comment (ThomasK33) [reply config/config.go:501]: "TranslatePatch" as struct is fine, as internally we perform translation operations (see the translate package in pkg).

**Comment by ThomasK33** [comment]
> Patch can both be a noun and a verb but I don't think it should be both in a single sentence. I would prefer a description such as "Defines patches to be applied to resources." or "Defines patches for resources", as patches are usually used for one thing- patching.

How does that align with Go doc conventions?

An example from the Go std lib: https://pkg.go.dev/io#Copy

**Review by FabianKramm — APPROVED** [review]

**Pull request merged**